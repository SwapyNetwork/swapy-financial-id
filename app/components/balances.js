import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { setInterval } from 'core-js';
import Styles from '../config/styles';
import WalletProvider from '../lib/wallet';
import IdentityProvider from '../lib/identity';

const getSwapyBalance = async identityAddress =>
  IdentityProvider
    .instance
    .getTokenBalance(identityAddress);

const getEthBalance = async walletAddress =>
  IdentityProvider.instance.getWeb3().eth.getBalance(walletAddress);

export default class Balances extends React.Component {
  constructor() {
    super();
    this.state = {
      ethBalance: undefined,
      swapyBalance: undefined,
    };
    setInterval(() => {
      this.loadBalances();
    }, 10000);
  }

  async loadBalances() {
    const walletAddress = WalletProvider.instance.getAddressString();
    let ethBalance = walletAddress ? await getEthBalance(WalletProvider.instance.getAddressString()) : undefined;
    ethBalance = ethBalance ?
      Number(IdentityProvider.instance.getWeb3().utils.fromWei(ethBalance)).toFixed(3) :
      ethBalance;

    const identityAddress = IdentityProvider.cachedIdentity ? IdentityProvider.cachedIdentity.address : undefined;
    let swapyBalance = identityAddress ? await getSwapyBalance(identityAddress) : undefined;
    swapyBalance = swapyBalance ?
      Number(IdentityProvider.instance.getWeb3().utils.fromWei(swapyBalance)).toFixed(3) :
      swapyBalance;

    this.setState({
      ethBalance,
      swapyBalance,
    });
  }

  render() {
    return (
      <View stles={styles.balances}>
        <Text style={styles.text}>
          {this.state.ethBalance ? `${this.state.ethBalance} ETH` : undefined}
        </Text>
        <Text style={styles.text}>
          {this.state.swapyBalance ? `${this.state.swapyBalance} SWAPY` : undefined}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  balances: {
  },
  text: {
    marginRight: 15,
    color: Styles.colors.white,
  },
});

