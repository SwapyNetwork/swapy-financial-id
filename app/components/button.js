/* @flow */

import React from 'react';
import { View, TouchableOpacity, Text, StyleSheet } from 'react-native';

import defaultStyles from '../config/styles';

type ButtonStyles = 'primary' | 'inverse' | 'transparent';
type Props = {
  label: string,
  onPress: () => any,
  buttonStyle?: ButtonStyles,
};

const Button = ({ label, onPress, buttonStyle = 'primary' }: Props) => (
  <View>
    <TouchableOpacity style={styles[buttonStyle]} onPress={onPress}>
      <Text style={styles[`${(buttonStyle: ButtonStyles)}Label`]}>{label}</Text>
    </TouchableOpacity>
  </View>
);

Button.defaultProps = { buttonStyle: 'primary' };

export default Button;

const styles = StyleSheet.create({
  primary: {
    marginVertical: 10,
    marginHorizontal: 20,
    padding: 10,
    borderRadius: 20,
    borderBottomWidth: 0,
    backgroundColor: '#00aeef',
    alignItems: 'center',
  },
  inverse: {
    marginVertical: 10,
    marginHorizontal: 20,
    padding: 10,
    borderRadius: 20,
    borderBottomWidth: 0,
    backgroundColor: '#00aeef',
    alignItems: 'center',
  },
  transparent: {
    marginVertical: 10,
    marginHorizontal: 20,
    padding: 10,
    borderRadius: 20,
    borderStyle: 'solid',
    borderWidth: 0.3,
    borderColor: defaultStyles.colors.gray,
    backgroundColor: 'transparent',
    alignItems: 'center',
  },
  primaryLabel: { color: 'white' },
  inverseLabel: { color: defaultStyles.colors.primary },
  transparentLabel: { color: defaultStyles.colors.gray },
});

