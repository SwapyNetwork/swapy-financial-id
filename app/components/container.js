/* @flow */

import * as React from 'react';
import { View, StyleSheet } from 'react-native';

type Props = {
  children? : React.Node,
};

const Container = (props: Props) => (
  <View style={styles.container} >
    {props.children}
  </View>
);

export default Container;

const styles = StyleSheet.create({
  container: {},
});
