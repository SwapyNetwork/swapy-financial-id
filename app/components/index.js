/* @flow */

import Container from './container';
import Button from './button';
import TextBox from './textBox';
import ProfileCard from './profileCard';
import Balances from './balances';

export {
  Container,
  Button,
  TextBox,
  ProfileCard,
  Balances,
};
