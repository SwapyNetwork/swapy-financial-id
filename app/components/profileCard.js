/* @flow */

import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import FeatherIcons from 'react-native-vector-icons/Feather';
import FontAwesomeIcons from 'react-native-vector-icons/FontAwesome';
import mainStyles from '../config/styles';

type Props = {
  name: string,
  email?: string,
  phone?: string,
  yearlyIncome?: number,
};

const ProfileCard = ({ name, email, phone, yearlyIncome }: Props) => (
  <View style={styles.card} >
    <FontAwesomeIcons name="user-circle-o" style={styles.avatar} size={130} color={mainStyles.colors.accent} />
    <View style={styles.properties}>
      <View style={styles.property}>
        <FeatherIcons style={styles.propertyIcon} name="user" size={32} color={mainStyles.colors.accent} />
        <Text style={styles.propertyText}>{name}</Text>
      </View>
      <View style={styles.property}>
        <FeatherIcons style={styles.propertyIcon} name="mail" size={32} color={mainStyles.colors.accent} />
        <Text style={styles.propertyText}>{email}</Text>
      </View>
      <View style={styles.property}>
        <FeatherIcons style={styles.propertyIcon} name="phone" size={32} color={mainStyles.colors.accent} />
        <Text style={styles.propertyText}>{phone}</Text>
      </View>
      <View style={styles.property}>
        <FeatherIcons style={styles.propertyIcon} name="star" size={32} color={mainStyles.colors.accent} />
        <Text style={styles.propertyText}>{'US$ '}{yearlyIncome}</Text>
      </View>
    </View>
  </View>
);

export default ProfileCard;
const styles = StyleSheet.create({
  card: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 10,
  },
  avatar: {
    padding: 10,
    marginTop: 10,
  },
  properties: {
    padding: 10,
    flex: 1,
  },
  property: {
    minHeight: 50,
    minWidth: '100%',
    flexDirection: 'row',
    paddingHorizontal: 30,
  },
  propertyIcon: {
    width: 40,
  },
  propertyText: {
    fontSize: 22,
    marginLeft: 30,
    color: mainStyles.colors.gray,
  },
  emptyProp: {
    margin: 5,
    padding: 2,
    height: 10,
    minWidth: '100%',
    backgroundColor: mainStyles.colors.gray,
  },
});
