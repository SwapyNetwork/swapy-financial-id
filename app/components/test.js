import * as React from 'react';

type Props = { /* ... */ };

type State = {
  count: number,
};

class MyComponent extends React.Component<Props, State> {
  state = {
    count: '123',
  };

  componentDidMount() {
    setInterval(() => {
      this.setState(prevState => ({
        count: prevState.count + 1,
      }));
    }, 1000);
  }

  render() {
    return <div>Count: {this.state.count}</div>;
  }
}

<MyComponent />;