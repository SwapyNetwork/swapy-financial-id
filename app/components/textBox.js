/* @flow */

import React from 'react';
import { ScrollView, StyleSheet, Text } from 'react-native';
import mainStyles from '../config/styles';

type Props = {
  bold: boolean,
  children: [],
};

const TextBox = (props: Props) => (
  <ScrollView style={styles.container} >
    <Text style={props.bold ? styles.textBold : styles.text}>
      {props.children}
    </Text>
  </ScrollView>
);

export default TextBox;

const styles = StyleSheet.create({
  container: {
    maxHeight: '70%',
    borderRadius: 10,
    borderStyle: 'solid',
    backgroundColor: mainStyles.colors.white,
    margin: '5%',
  },
  text: {
    padding: '3%',
    height: '100%',
    overflow: 'visible',
  },
  textBold: {
    padding: '3%',
    height: '100%',
    fontSize: 26,
    fontWeight: 'bold',
    overflow: 'visible',
  },
});
