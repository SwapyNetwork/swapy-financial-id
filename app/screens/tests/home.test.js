/* global it, expect, describe, jest */
/* eslint-disable object-curly-newline */
/* eslint-disable global-require */

import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Text, ActivityIndicator } from 'react-native';

import Home from '../home';
import { Button } from '../../components';

jest.mock('../../lib/wallet', () => ({
  instance: {
    getPrivateKeyString: () => '0x123456123456123456',
  },
  initWalletFromStorage: jest.fn()
    .mockImplementationOnce(async () => true)
    .mockImplementationOnce(async () => false)
    .mockImplementation(async () => true),
}));

jest.mock('../../lib/identity', () => ({
  instance: {
    getTreeData: async () => ({ id: 'test', hash: '1283ashdh1238' }),
  },
  retrieveIdentity: jest.fn()
    .mockImplementation(async () => ({ id: 'test', hash: '1283ashdh1238' })),
  factoryUserData: {},
}));

jest.mock('react-native-splash-screen', () => ({
  hide: () => null,
}));

describe('<Home />', async () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Home />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('displays "create" and "import" buttons when there\'s no wallet', async () => {
    const wrapper = mount(<Home />);

    process.nextTick(async () => { // Ensures componentDidMount was called
      expect(require('../../lib/identity').retrieveIdentity).toHaveBeenCalledTimes(2);
      expect(wrapper.find(Button).at(0).props().label).toEqual('Create new wallet');
      expect(wrapper.find(Button).at(1).props().label).toEqual('Import existing wallet');
    });
  });

  it('renders the ActivityIndicator when is loading', async () => {
    const wrapper = mount(<Home />);

    expect(wrapper.find(ActivityIndicator)).toBeTruthy();
    process.nextTick(async () => { // Ensures componentDidMount was called
      expect(require('../../lib/identity').retrieveIdentity).toHaveBeenCalledTimes(2);
      expect(wrapper.find(Button).at(1).props().label).toEqual('Import existing wallet');
    });
  });
});
