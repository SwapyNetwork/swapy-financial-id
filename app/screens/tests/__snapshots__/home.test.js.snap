// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#00aeef",
        "fontSize": 48,
        "fontWeight": "bold",
        "top": 40,
      }
    }
  >
    Welcome
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#00aeef",
        "fontSize": 16,
        "marginBottom": 40,
        "marginTop": 25,
      }
    }
  >
    to Swapy Financial ID
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../app/assets/logo.png",
      }
    }
    style={
      Object {
        "height": 150,
        "marginBottom": 40,
        "width": 150,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 50,
        "paddingRight": 50,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#4a4a4a",
          "fontSize": 16,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Your Financial ID will be linked to a ETH Wallet
    </Text>
    <ActivityIndicator
      animating={true}
      color="#00aeef"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
